ui/cypress/e2e/collectors.test.ts

```diff
@@ -165,5 +170,99 @@ describe('Collectors', () => {
       cy.getByTestID('resource-card').should('contain', secondTelegraf)
       cy.getByTestID('resource-card').should('not.contain', thirdTelegraf)
     })
+
+    it('can sort telegraf configs by name', () => {
+      const telegrafs = ['b', 'a', 'c']
+      const [firstTelegraf, secondTelegraf, thirdTelegraf] = telegrafs
+      const description = 'Config Description'
+
+      cy.get('@org').then(({id}: Organization) => {
+        cy.createTelegraf(firstTelegraf, description, id, bucket)
+        cy.createTelegraf(secondTelegraf, description, id, bucket)
+        cy.createTelegraf(thirdTelegraf, description, id, bucket)
+      })
+
+      cy.reload()
+
+      // test the order
+      // cy.getByTestID('resource-card').should('have.length', 3)
+      cy.getByTestID('collector-card--name')
+        .not('a')
+        .should('have.length', 3)
+
+      // test to see if telegrafs are initially sorted by name
+      telegrafs.sort()
+
+      cy.getByTestID('collector-card--name')
+        .each((val, index) => {
+          expect(val.text()).to.equal(telegrafs[index])
+        })
+        .then(() => {
+          cy.getByTestID('resource-list--sorter')
+            .contains('Name')
+            .click()
+          telegrafs.reverse()
+        })
+        .then(() => {
+          cy.getByTestID('collector-card--name').each((val, index) => {
+            expect(val.text()).to.equal(telegrafs[index])
+          })
+        })
+    })
+
+    it('can sort telegraf configs by bucket', () => {
+      const telegrafs = ['telegraf_a', 'telegraf_b', 'telegraf_c']
+      const [firstTelegraf, secondTelegraf, thirdTelegraf] = telegrafs
+      const description = 'Config Description'
+      const bucketz = ['MO_buckets', 'EZ_buckets', 'Bucky']
+      const [firstBucket, secondBucket, thirdBucket] = bucketz
+
+      cy.get('@org').then(({id}: Organization) => {
+        cy.createTelegraf(firstTelegraf, description, id, firstBucket)
+        cy.createTelegraf(secondTelegraf, description, id, secondBucket)
+        cy.createTelegraf(thirdTelegraf, description, id, thirdBucket)
+      })
+
+      cy.reload()
+
+      // test the order
+      // find all the cf-resource-card--meta-item without children
+      cy.getByTestID('cf-resource-card--meta-item').should('have.length', 6)
```

Comment:
since we are using built-in testIDs from the `@clockface` dependencies, I couldn't figure out a way to fully customize the selectors. I tried using `not`, `filter`, chaining `get`, `contains`, etc... all without reaching the intended purpose. I would love to work on fixing this since this is shouldn't be merged onto master

Reply:
you should be able to pass arbitrary testIDs / overwrite the default testIDs for any clockface component.